src/seals/util.py

```diff
@@ -67,26 +75,31 @@ def _step_pad(self, action):
         """
         if self.previous_done:
             self.previous_done = False
+            reset_obs, reset_info_dict = self.env.reset()
+            info = {"reset_info_dict": reset_info_dict}
             # This transition will only reset the environment, the action is ignored.
-            return self.env.reset(), self.reset_reward, False, {}
+            return reset_obs, self.reset_reward, False, False, info
 
-        obs, rew, done, info = self.env.step(action)
-        if done:
+        obs, rew, terminated, truncated, info = self.env.step(action)
+        if terminated:
             self.previous_done = True
-        return obs, rew, False, info
+        return obs, rew, False, truncated, info
 
-    def _step_discard(self, action):
+    def _step_discard(
+        self, action: WrapperActType
+    ) -> tuple[ObsType, SupportsFloat, bool, bool, dict[str, Any]]:
         """When done=True, returns done=False instead and automatically resets.
 
         When an automatic reset happens, the observation from reset is returned,
         and the overridden observation is stored in
         `info["terminal_observation"]`.
         """
-        obs, rew, done, info = self.env.step(action)
-        if done:
+        obs, rew, terminated, truncated, info = self.env.step(action)
```

Comment:
There's a subtle change in logic here that might need additional thought.

Previously, `done` would represent both hitting a terminal state (which we generally want to remove in seals from what I understand), and hitting a time limit.

Now, terminal state is `terminated`, and time limit is `truncated`


The standard conversion recommendation is changing `if done:` to `if terminated or truncated:`, but it's possible it needs to be different here - please double check that

Reply:
Discussion continued [here](https://github.com/HumanCompatibleAI/seals/pull/73/files#r1291362553)